theatre/shared/src/utils/types.ts

```diff
@@ -11,7 +11,8 @@ export type SerializableMap<
   Primitives extends SerializablePrimitive = SerializablePrimitive,
 > = {[Key in string]?: SerializableValue<Primitives>}
 
-export type SerializablePrimitive = string | number | boolean
+// TODO: I'm sure we can make this type make more sense
+export type SerializablePrimitive = string | number | boolean | {}
```

Comment:
I'm surprised `| {}` is not breaking anything. A `Primitive` is supposed to be, well, anything other than `Compound` (which itself includes `Enum`s). Adding `{}` to `Primitive` makes it accept anything iiuc. Maybe instead of `{}` we could hard-code `{r: number, g: number, b: number, a: number}` here until a better solution is found?

Reply:
Yeah, ideally it would be a type that is any object where object.type !== 'compound', however that kind of type doesn't seem to be possible to construct with typescript.

The only trick that comes to mind is this, but it's ugly:
```ts
type SerializablePrimitive<T> = string | number | boolean | (T extends {type: 'compound'} ? never : T);

const badStuff = {
    type: 'compound',
    foo: 3,
} as const

const goodStuff = {
    type: 'literallyanythingelse',
    foo: 3,
} as const

function serializeStuff<T>(giveMeStuff: SerializablePrimitive<T>) {
    // ...
} 

serializeStuff(badStuff) // ⚠️
serializeStuff(goodStuff) // 💏
```

Reply:
I think for now I'll go with the hard-coded one, but I'll leave a comment to fix it.